File: VB/SecuredExportExample.Module/SecurityObjects/ExtendedSecurityRole.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.Security
Imports System.Collections.Generic
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy

Namespace SecuredExportExample.Module.SecurityObjects
    <DefaultClassOptions, ImageName("BO_Role")> _
    Public Class ExtendedSecurityRole
        Inherits PermissionPolicyRole

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property CanExport() As Boolean
            Get
                Return GetPropertyValue(Of Boolean)("CanExport")
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("CanExport", value)
            End Set
        End Property
'        
'        protected override IEnumerable<IOperationPermission> GetPermissionsCore() {
'            List<IOperationPermission> result = new List<IOperationPermission>();
'            result.AddRange(base.GetPermissionsCore());
'            if (CanExport) result.Add(new ExportPermission());
'            return result;
'        }
    End Class
End Namespace

File: CS/SecuredExportExample.Module/SecurityObjects/ExtendedSecurityRole.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Security;
using System.Collections.Generic;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;

namespace SecuredExportExample.Module.SecurityObjects {
    [DefaultClassOptions, ImageName("BO_Role")]
    public class ExtendedSecurityRole : PermissionPolicyRole {
        public ExtendedSecurityRole(Session session) : base(session) { }
        public bool CanExport {
            get { return GetPropertyValue<bool>("CanExport"); }
            set { SetPropertyValue<bool>("CanExport", value); }
        }
        /*
        protected override IEnumerable<IOperationPermission> GetPermissionsCore() {
            List<IOperationPermission> result = new List<IOperationPermission>();
            result.AddRange(base.GetPermissionsCore());
            if (CanExport) result.Add(new ExportPermission());
            return result;
        }*/
    }
}
